### ISSUE TITLE
Enable passing config package name to tslint CLI

### ISSUE THREAD

**Issue by martijnthe** [palantir/tslint#3503]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.6.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
// Not relevant
```

with `tslint.json` configuration:

```json
"Not relevant"
```

#### Actual behavior

```
$ yarn tslint -c my-tslint-config-package-name -p .
Invalid option for configuration: my-tslint-config-package-name
error Command failed with exit code 1.
```
#### Expected behavior

I would expect there to be an CLI option to specify the tsconfig package name, but it seems it only takes a file path?

Perhaps `-c` could first assume it gets a file path and if that doesn't exist, try resolving it as a module name.

**Comment by ajafff** [comment]
I also thought about that. Unfortunately there is no way to tell other tools what configuration to use. For example editor plugins need to look up the configuration on the file sytem. That way they don't know what configuration you pass on the command line.

**Comment by martijnthe** [comment]
@ajafff Yeah, that's a good point, but that's already true of the existing `-c` option.

Personally, I'm mostly interested in this to reduce the number of `xyz-config.js[on]` files. I've got a mono repo with dozens of packages, many of them having a `tslint.json`, `tsconfig.json`, webpack config, eslint config, etc. etc. All of those files are more or less the same (just pointing to a package with the actual config...)

I guess this could also help to reduce the config file mania: https://github.com/palantir/tslint/issues/3031

**Comment by adidahiya** [comment]
@martijnthe if you have a monorepo, you might be interested in how I managed this in blueprint: https://github.com/palantir/blueprint/blob/3579ce26bacfa61193721c8d97505fff32e00b97/packages/core/package.json#L22

**Comment by martijnthe** [comment]
@adidahiya thanks, so passing the path to a package works, but not passing just the package name?

**Comment by adidahiya** [comment]
@martijnthe correct; we don't resolve the `-c` string as a path to a node module. But I think my approach at least helps you reduce the number of config files from N to 1?

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
